File: sdk/nodejs/get_product_market_data_example.js
var StockX = require('stock_x_api');

var exampleConstants = new ExampleConstants();

var defaultClient = StockX.ApiClient.instance;
	defaultClient.basePath = 'https://gateway.stockx.com/stage';

var apiKey = defaultClient.authentications['api_key'];
	apiKey.apiKey = exampleConstants.AWS_API_KEY;

var stockx = new StockX.StockXApi();

var login = new StockX.LoginRequest();
	login.email = exampleConstants.STOCKX_USERNAME;
	login.password = exampleConstants.STOCKX_PASSWORD;

stockx.loginWithHttpInfo(login)
	.then(res => {
		defaultClient.defaultHeaders['jwt-authorization'] = res.response.headers['jwt-authorization'];
		return res.data.Customer;
	}).then(customer => {
		// Lookup the supplied product by its product ID
		return stockx.getProductMarketData('7f9c67ff-2cd4-4e27-8b64-518900a1bc0b', 'air-jordan-1-retro-high-off-white-chicago');	
	}).then(search => {
		console.log('Product market data lookup successful: ' + JSON.stringify(search));
	}).catch(err => {
		console.log('Error performing product market data lookup: ' + err);
	});

File: sdk/python/src/get_product_market_data_example.py
from __future__ import print_function
import time
import io_stockx
from example_constants import ExampleConstants
from io_stockx.rest import ApiException
from pprint import pprint

# Configure API key authorization: api_key
configuration = io_stockx.Configuration()

configuration.host = "https://gateway.stockx.com/stage"
configuration.api_key['x-api-key'] = ExampleConstants.AWS_API_KEY

# create an instance of the API class
stockx = io_stockx.StockXApi(io_stockx.ApiClient(configuration))
login = io_stockx.LoginRequest(email=ExampleConstants.STOCKX_USERNAME, password=ExampleConstants.STOCKX_PASSWORD)

try:
    # Attempts to log the user in with a username and password.
    api_response = stockx.login(login)

    result = stockx.get_product_market_data('7f9c67ff-2cd4-4e27-8b64-518900a1bc0b')

    pprint(result)
except ApiException as e:
    print("Exception when calling StockXApi->get_product_market_data: %s\n" % e)
